tests/application/job_service/test_job.py

```diff
@@ -189,3 +201,15 @@ async def test_if_job_ran_successful_then_log_runtime_of_job(self):
 
         assert any("Job running time" in logged_msg[1][0] for logged_msg in info_logger.mock_calls)
         assert any(f"{timedelta(seconds=15)}" in logged_msg[1][0] for logged_msg in info_logger.mock_calls)
+
+
+def get_random_retry_error_code():
```

Comment:
I completely get why you did that, I'm personally just not a big fan of having randomized test, especially with our already shaky pipeline. Adding another uncertainty maybe isn't the best thing? So I'd probably opt for just testing that SLOT_EMPTY is retried and TRANFER_NHEADER isn't. wdyt?

Reply:
That was my thought too but then I thought that if we modify the list of retryable errors, then we may missed the one we statically specified in the tests. BUT you are right, we can use two errors that for sure:

- 1 should always be retried like ..SLOT_EMPTY
- 1 should never be retried like ALREADY_OPEN_UNLOCK_CODE_REQUEST
?

Reply:
Yes, great. Let's do it like that.

Reply:
Done